src/AutoMapper/TypeExtensions.cs

```diff
@@ -36,18 +36,35 @@ public static Type CreateType(this TypeBuilder type)
 
         public static IEnumerable<MemberInfo> GetAllMembers(this Type type)
         {
-            while (true)
+            if (type.IsInterface())
```

Comment:
Ok, so what would you suggest? Adding another extensionmethod to the TypeExtensions class like 
`public static IEnumerable<MemberInfo> GetAllMembersIncludingInterfaces( this Type type)`

Any precise suggestions are very welcome. I did not develop this feature, I just made a bugfix for something we need in production ;-)